common/djangoapps/student/forms.py

```diff
@@ -396,27 +385,48 @@ def clean_email(self):
             )
         return email
 
-    def clean_year_of_birth(self):
-        """
-        Parse year_of_birth to an integer, but just use None instead of raising
-        an error if it is malformed
-        """
-        try:
-            year_str = self.cleaned_data["year_of_birth"]
-            return int(year_str) if year_str is not None else None
-        except ValueError:
-            return None
-
-    @property
-    def cleaned_extended_profile(self):
-        """
-        Return a dictionary containing the extended_profile_fields and values
-        """
-        return {
-            key: value
-            for key, value in self.cleaned_data.items()
-            if key in self.extended_profile_fields and value is not None
-        }
+    def clean_org_name(self):
+        """ Enforce organization related field conditions"""
+        clean_org_name = self.cleaned_data.get("org_name")
+        clean_org_size = self.cleaned_data.get("org_size")
+        clean_org_type = self.cleaned_data.get("org_type")
+
+        possible_org_types = OrgSector.objects.values_list('code', flat=True)
+        possible_org_sizes = TotalEmployee.objects.values_list('code', flat=True)
+
+        if clean_org_type and clean_org_type not in possible_org_types:
+            raise forms.ValidationError(_('Invalid org type option provided'))
+            return clean_org_name
```

Comment:
Lets verify, is return statement reachable. Also can we avoid short form of words in messages like `org`.

Please verify both cases for following block as well

Reply:
You were right, that was another scenario I was thinking about where instead of `raising` the error, we put it in a list. I have removed the extra `return`s